dotcom-rendering/src/amp/components/Elements.tsx

```diff
@@ -21,6 +22,79 @@ import { VideoYoutubeBlockComponent } from './elements/VideoYoutubeBlockComponen
 import { YoutubeBlockComponentAMP } from './elements/YoutubeBlockComponentAMP';
 import { Expandable } from './Expandable';
 
+/**
+ * Elements we support on AMP -
+ *
+ * Elements with isMandatory property should only be included if we render
+ * them when isMandatory is true
+ */
+const AMP_SUPPORTED_ELEMENTS = [
+	'model.dotcomrendering.pageElements.AudioAtomBlockElement',
+	'model.dotcomrendering.pageElements.BlockquoteBlockElement',
+	'model.dotcomrendering.pageElements.ChartAtomBlockElement',
+	'model.dotcomrendering.pageElements.CommentBlockElement',
+	'model.dotcomrendering.pageElements.ContentAtomBlockElement',
+	'model.dotcomrendering.pageElements.DisclaimerBlockElement',
+	// We do not support EmbedBlockElement's when they are mandatory
+	// 'model.dotcomrendering.pageElements.EmbedBlockElement',
+	'model.dotcomrendering.pageElements.GenericAtomBlockElement',
+	'model.dotcomrendering.pageElements.GuideAtomBlockElement',
+	'model.dotcomrendering.pageElements.GuVideoBlockElement',
+	'model.dotcomrendering.pageElements.ImageBlockElement',
+	'model.dotcomrendering.pageElements.InteractiveAtomBlockElement',
+	// We do not support InteractiveBlockElement's when they are mandatory
+	// 'model.dotcomrendering.pageElements.InteractiveBlockElement',
+	'model.dotcomrendering.pageElements.ProfileAtomBlockElement',
+	'model.dotcomrendering.pageElements.PullquoteBlockElement',
+	'model.dotcomrendering.pageElements.QABlockElement',
+	'model.dotcomrendering.pageElements.RichLinkBlockElement',
+	'model.dotcomrendering.pageElements.SoundcloudBlockElement',
+	'model.dotcomrendering.pageElements.SubheadingBlockElement',
+	'model.dotcomrendering.pageElements.TextBlockElement',
+	'model.dotcomrendering.pageElements.TimelineBlockElement',
+	'model.dotcomrendering.pageElements.TweetBlockElement',
+	'model.dotcomrendering.pageElements.VideoVimeoBlockElement',
+	'model.dotcomrendering.pageElements.VideoYoutubeBlockElement',
+	'model.dotcomrendering.pageElements.YoutubeBlockElement',
+];
+
+export const isAmpSupported = (
+	format: CAPIFormat,
+	tags: TagType[],
+	elements: CAPIElement[],
+	switches: Switches,
+	main: string,
+): boolean => {
+	if (format.design === 'InteractiveDesign') {
+		const hasAmpInteractiveTag = tags.some(
+			(tag) => tag.id === 'tracking/platformfunctional/ampinteractive',
+		);
+		if (!hasAmpInteractiveTag) return false;
+	}
+
+	if (format.design === 'ArticleDesign') {
```

Comment:
Hmm, sorry I've been bouncing back and forth on this, but I think we do actually want to do something like
`tags.includes('type/article')` instead of `format.design === 'ArticleDesign'` to deal with all the different  types of articles such as Analysis, Explainers, etc

Reply:
Now updated!